src/debugAdapter/goDebug.ts

```diff
@@ -503,11 +503,21 @@ class GoDebugSession extends DebugSession {
 		return path.replace(this.delve.program, this.delve.remotePath).split(this.localPathSeparator).join(this.remotePathSeparator);
 	}
 
-	protected toLocalPath(path: string): string {
+	protected toLocalPath(pathToConvert: string): string {
 		if (this.delve.remotePath.length === 0) {
-			return this.convertDebuggerPathToClient(path);
+			return this.convertDebuggerPathToClient(pathToConvert);
 		}
-		return path.replace(this.delve.remotePath, this.delve.program).split(this.remotePathSeparator).join(this.localPathSeparator);
+
+		// Fix for https://github.com/Microsoft/vscode-go/issues/1178
+		// When the pathToConvert is under GOROOT, replace the remote GOROOT with local GOROOT
+		if (!pathToConvert.startsWith(this.delve.remotePath)) {
+			let index = pathToConvert.indexOf('src');
```

Comment:
This is supposed to be `/src/` in the goroot, right? It could match a different src, like `/Users/src/goroot/src/...`

Reply:
Updated to match `/src/`